airbyte-cdk/python/airbyte_cdk/sources/file_based/file_based_source.py

```diff
@@ -165,24 +163,40 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
                     if (state_manager and catalog_stream)
                     else None
                 )
+                # Like state_manager, `sync_mode` may be None during `check`
+                sync_mode = self._get_sync_mode_from_catalog(stream_config.name)
+
                 self._validate_input_schema(stream_config)
 
-                sync_mode = self._get_sync_mode_from_catalog(stream_config.name)
+                assert (
+                    hasattr(self, "_concurrency_level") and self._concurrency_level is not None
+                ), "Concurrency level is not set for the source. Please set the _concurrency_level attribute."
+                assert issubclass(
+                    self.cursor_cls, AbstractConcurrentFileBasedCursor
+                ), "Cursor class must be a subclass of AbstractConcurrentFileBasedCursor."
+
+                if sync_mode is None:  # Some hacky stuff for the check case
+                    cursor = self.cursor_cls(
+                        stream_config,
+                        stream_config.name,
+                        None,
+                        stream_state,
+                        self.message_repository,
+                        state_manager,  # None, but we don't use it in this case
+                        CursorField(DefaultFileBasedStream.ab_last_mod_col),
+                    )
+                    # TODO: should use stream facade
```

Comment:
flagging the todo. Can we either use the stream facade or remove the todo if it doesn't matter?

I suspect we can't just use the facade because of the weird set up for the check